File: linux/alertaway/src/Python_Perl_test.pl
# testing QueueManager with messages from python process
# as well as now sending/receiving pickled python messages
use Data::Dumper;
use strict;
use tools qw (:debug);
use cfg;
use QueueManager;
use db;
use filterPrint;
use feature 'state';
use constant QUEUE_TIMEOUT => 120;

sub task
{
    
    my ($trace_in, $api, $evaluate_pid) = @_;
    my  $SendQueue= QueueManager::SendQueue({python_reader => 1});
    $SendQueue->enqueue({name => 'from perl living_room_light', value => 'ON'});  # turn on
    {
        my $ReceiveQueue = QueueManager::ReceiveQueue();
        $ReceiveQueue->enqueue({name => 'perl living_room_light', value => 'ON'}); # IS on
    }

    my $ReceiveQueue = QueueManager::ReceiveQueue({reader => 1});
    sleep(4);
    while (1)
    {
       my $message = $ReceiveQueue->dequeue(); # just used mostly for an event
       if (!$message)
       {
        printf("timed out");
       }
       else
       {
        printf("test.message[%s]\n", Dumper $message);
       }
    }
}
task();

File: linux/alertaway/src/Python_perl_test.py
from ipcqueue import posixmq
import time
#testing Perl QueueManger.pm to have it handle python pickled messages  as well as perl "Storable"
#send to perl example 
# rq= posixmq.Queue('/ReceiveQueue')
# while True:
#     print("sending message")
#     rq.put({"name": "python water_sensor_2", "value": "foobar to you"})
#     print(rq.qsize())
#     time.sleep(4)

receive from perl example
rq= posixmq.Queue('/SendQueue')
while True:
    print("sending message")
    try:
        msg = rq.get(timeout=None)
    except:
        print("problem getting message")
    print(rq.qsize())
    print(msg)
